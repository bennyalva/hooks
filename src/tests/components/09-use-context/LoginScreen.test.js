import React from 'react'
const { mount } = require("enzyme");
const { UserContext } = require("../../../components/09-useContex/UserContext");
const { LoginScreen } = require("../../../components/09-useContex/LoginScreen");

describe('Tests loginScreen', () => {
    const setUser = jest.fn();
    const wrapper = mount(<UserContext.Provider value={{
        setUser
    }}>
        <LoginScreen />
    </UserContext.Provider>)
   test('should show correctly', () => {
       expect(wrapper).toMatchSnapshot();
   });
    test('should call setUser', () => {
        wrapper.find('button').prop('onClick')();
       expect(setUser).toHaveBeenCalledWith({id: 123456, name: 'rey'});
    });
    
});
