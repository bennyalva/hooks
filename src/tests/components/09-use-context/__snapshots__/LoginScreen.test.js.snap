// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests loginScreen should show correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
